src/macro_expansion.jl

```diff
@@ -275,7 +275,13 @@ function expand_macro(ctx, ex)
     # age changes concurrently.
     #
     # TODO: Allow this to be passed in
-    if hasmethod(macfunc, Tuple{typeof(mctx), typeof.(raw_args)...}; world=ctx.macro_world)
+    # TODO: hasmethod always returns false for our `typemax(UInt)` meaning
+    # "latest world," which we shouldn't be using.
+    has_new_macro = ctx.macro_world === typemax(UInt) ?
+        hasmethod(macfunc, Tuple{typeof(mctx), typeof.(raw_args)...}) :
+        hasmethod(macfunc, Tuple{typeof(mctx), typeof.(raw_args)...}; world=ctx.macro_world)
```

Comment:
Ok, let's leave the special case for now but just be a little more explicit:

```suggestion
    # TODO: We use `typemax(UInt)` to mean "latest world" in some places, but
    # we should remove that and the following special case.
    macro_world = ctx.macro_world === typemax(UInt) ? Base.get_world_counter() : ctx.macro_world
    has_new_macro = hasmethod(macfunc, Tuple{typeof(mctx), typeof.(raw_args)...}; world=macro_world)
```